services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/ShieldSprites.java

```diff
@@ -0,0 +1,128 @@
+package com.mapbox.api.directions.v5.models;
+
+import androidx.annotation.NonNull;
+
+import com.google.auto.value.AutoValue;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.TypeAdapter;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * ShieldSprites.
+ */
+@AutoValue
+public abstract class ShieldSprites implements Serializable {
+
+  /**
+   * Create a new instance of this class by using the {@link ShieldSprites.Builder} class.
+   *
+   * @return {@link ShieldSprites.Builder} for creating a new instance
+   */
+  public static Builder builder() {
+    return new AutoValue_ShieldSprites.Builder();
+  }
+
+  /**
+   * Create a new instance of this class by passing in a formatted valid JSON String.
+   *
+   * @param json a formatted valid JSON string defining a shield sprite
+   * @return a new instance of this class defined by the values passed inside this static factory
+   *   method
+   */
+  @NonNull
+  public static ShieldSprites fromJson(@NonNull String json) {
+    List<ShieldSprite> sprites = new ArrayList<>();
+    GsonBuilder gson = new GsonBuilder();
+    JsonObject jsonObject = gson.create().fromJson(json, JsonObject.class);
```

Comment:
Since `gson.create().fromJson` throws a JSONSyntaxException in case of invalid representation for an object, should we wrap this entire piece inside a `try/catch`?
cc @Guardiola31337

Reply:
If that's the case the exception will be thrown. We don't need to `try/catch` unless we want to fail silently or do something in addition before throwing the exception.

Reply:
Okay I think it makes sense to throw exception and let the SDK/app user handle the exception upstream.